package.json

```diff
@@ -24,18 +28,26 @@
   "jest": {
     "verbose": true
   },
+  "engines": {
+    "node": ">=12.14.0"
+  },
   "dependencies": {
-    "virtualfs": "^2.2.0"
+    "threads": "1.4.0",
+    "virtualfs": "^2.2.0",
+    "errno": "^0.1.7"
   },
   "devDependencies": {
-    "@types/jest": "^24.9.1",
+    "@types/jest": "^25.0.0",
     "@types/node": "^13.13.1",
```

Comment:
The `"@types/node"` provides the fs type: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/fs.d.ts.

This should mean we do not need to create our own Filesystem interface.

Furthermore as long as we use the `fs` as a type, and not as an implementation. The typescript compilation should remove all references when it generates the JS and thus we won't have any portability problems.

Reply:
this is now resolved by using the fs type on lower directory and an any type on the upper directory